compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/ExposedDropdownMenu.skiko.kt

```diff
@@ -0,0 +1,713 @@
+            focusedLeadingIconColor = focusedLeadingIconColor,
+            unfocusedLeadingIconColor = unfocusedLeadingIconColor,
+            disabledLeadingIconColor = disabledLeadingIconColor,
+            errorLeadingIconColor = errorLeadingIconColor,
+            focusedTrailingIconColor = focusedTrailingIconColor,
+            unfocusedTrailingIconColor = unfocusedTrailingIconColor,
+            disabledTrailingIconColor = disabledTrailingIconColor,
+            errorTrailingIconColor = errorTrailingIconColor,
+            focusedLabelColor = focusedLabelColor,
+            unfocusedLabelColor = unfocusedLabelColor,
+            disabledLabelColor = disabledLabelColor,
+            errorLabelColor = errorLabelColor,
+            focusedPlaceholderColor = focusedPlaceholderColor,
+            unfocusedPlaceholderColor = unfocusedPlaceholderColor,
+            disabledPlaceholderColor = disabledPlaceholderColor,
+            errorPlaceholderColor = errorPlaceholderColor,
+            focusedPrefixColor = focusedPrefixColor,
+            unfocusedPrefixColor = unfocusedPrefixColor,
+            disabledPrefixColor = disabledPrefixColor,
+            errorPrefixColor = errorPrefixColor,
+            focusedSuffixColor = focusedSuffixColor,
+            unfocusedSuffixColor = unfocusedSuffixColor,
+            disabledSuffixColor = disabledSuffixColor,
+            errorSuffixColor = errorSuffixColor,
+        )
+}
+
+@Suppress("ComposableModifierFactory")
+@Composable
+private fun Modifier.expandable(
+    expanded: Boolean,
+    onExpandedChange: () -> Unit,
+    menuDescription: String = getString(Strings.ExposedDropdownMenu),
+    expandedDescription: String = getString(Strings.MenuExpanded),
+    collapsedDescription: String = getString(Strings.MenuCollapsed),
+) = pointerInput(Unit) {
+    awaitEachGesture {
+        // Must be PointerEventPass.Initial to observe events before the text field consumes them
+        // in the Main pass
+        awaitFirstDown(pass = PointerEventPass.Initial)
+        val upEvent = waitForUpOrCancellation(pass = PointerEventPass.Initial)
+        if (upEvent != null) {
+            onExpandedChange()
+        }
+    }
+}.semantics {
+    stateDescription = if (expanded) expandedDescription else collapsedDescription
+    contentDescription = menuDescription
+    onClick {
+        onExpandedChange()
+        true
+    }
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+private fun updateHeight(
+    windowInfo: WindowInfo,
+    coordinates: LayoutCoordinates?,
+    verticalMarginInPx: Int,
+    onHeightUpdate: (Int) -> Unit
+) {
+    coordinates ?: return
+    val boundsInWindow = coordinates.boundsInWindow()
+    val visibleWindowBounds = windowInfo.size.toIntRect()
+    val heightAbove = boundsInWindow.top - visibleWindowBounds.top
+    val heightBelow = visibleWindowBounds.height - boundsInWindow.bottom
+    onHeightUpdate(max(heightAbove, heightBelow).toInt() - verticalMarginInPx)
```

Comment:
```suggestion
    onHeightUpdate((max(heightAbove, heightBelow).toInt() - verticalMarginInPx).coerceAtLeast(0))
```

Reply:
> boundsInWindow is supposed to be always inside visibleWindowBounds

It is implementation details of the other part of code. This part of code shouldn't depend on it, and should return a correct height. A negative height isn't correct by its semantics.

Not highly important, but it can help to avoid future bugs.

>  it will be corrected on height limit modifier

It passes straight to `heightIn(max = menuHeight.toDp()`, which can have a restriction to have non-negative value.

Reply:
After looking at `boundsInRoot()`, it seems that it coerces the values by the window bounds.

I am not sure if it is its contract, or a mistake 🤔. `boundsInRoot` on the other hand have more  sophisticated behavior:
<img width="588" alt="image" src="https://github.com/JetBrains/compose-multiplatform-core/assets/5963351/89a7fa6d-953a-4a8b-be2f-f878c4fbc4ed">

If it would be a contract, than indeed, we wouldn't have to to make coercing ourselves here.

Anyway, let's stick to something, I will support any decision. This isn't important.

Reply:
I'd avoid introducing extra logic (comparing to android) if it doesn't solve anything at the moment